mm2src/coins/eth/fee_estimation/eip1559/mod.rs

```diff
@@ -0,0 +1,89 @@
+//! Provides estimations of base and priority fee per gas or fetch estimations from a gas api provider
+pub mod block_native;
+pub mod infura;
+pub mod simple;
+
+use ethereum_types::U256;
+use url::Url;
+
+const FEE_PER_GAS_LEVELS: usize = 3;
+
+/// Indicates which provider was used to get fee per gas estimations
+#[derive(Clone, Debug)]
+pub enum EstimationSource {
+    /// filled by default values
+    Empty,
+    /// internal simple estimator
+    Simple,
+    Infura,
+    Blocknative,
+}
+
+impl ToString for EstimationSource {
+    fn to_string(&self) -> String {
+        match self {
+            EstimationSource::Empty => "empty".into(),
+            EstimationSource::Simple => "simple".into(),
+            EstimationSource::Infura => "infura".into(),
+            EstimationSource::Blocknative => "blocknative".into(),
+        }
+    }
+}
+
+impl Default for EstimationSource {
+    fn default() -> Self { Self::Empty }
+}
+
+enum PriorityLevelId {
+    Low = 0,
+    Medium = 1,
+    High = 2,
+}
+
+/// Supported gas api providers
+#[derive(Clone, Deserialize)]
+pub enum GasApiProvider {
+    Infura,
+    Blocknative,
+}
```

Comment:
could be useful for my note above.

Reply:
backlogged in https://github.com/KomodoPlatform/komodo-defi-framework/issues/2207